File: agency_scraper/agency_scraper/spiders/koto.py
import scrapy
from datetime import datetime


class KotoSpider(scrapy.Spider):
    name = "koto"
    allowed_domains = ["koto.studio"]
    start_urls = ["https://koto.studio/work"]

    def parse(self, response):
        projects = response.css('a.work-row-thumb.tile')  # Target each project card

        for project in projects:
            title = project.css('h2::text').get(default='Untitled').strip()
            href = project.css('::attr(href)').get()
            video = project.css('::attr(data-work-page-thumbnail-video)').get()
            
            yield {
                "title": title,
                "description": "",  # Optional: we don't have it here yet
                "project_page_url": response.urljoin(href),
                "media_url": [video] if video else [],
                "tags": []  # Empty for now, you can add manually later if needed
            }

File: old_backend/scrapers/koto.js
const axios = require('axios');
const cheerio = require('cheerio'); // 🧼 jQuery-like HTML parser

async function scrapeKoto() {
  const url = 'https://koto.studio/work';
  const { data: html } = await axios.get(url); // 🛰 Fetch page
  const $ = cheerio.load(html);                // 🧼 Load DOM

  const projects = [];

  // Target each project item on the page
  $('.project-item').each((_, el) => {
    // Extract the project URL (href)
    const href = $(el).find('a').attr('href');
    
    // Extract the video URL (data-work-page-thumbnail-video)
    const videoUrl = $(el).find('a').attr('data-work-page-thumbnail-video');

    // Extract the title
    const title = $(el).find('.work-meta h2').text().trim();

    // Push project data into the array
    projects.push({
      title,
      url: `https://koto.studio${href}`,
      videoUrl,
      source: 'koto.studio'
    });
  });

  return projects;
}

module.exports = scrapeKoto;
